templates/cicd/pipeline_cfn.yml

```diff
@@ -8,11 +8,13 @@
 AWSTemplateFormatVersion: '2010-09-09'
 Description: CodePipeline for {{$.AppName}}
 Parameters:
-  # TODO: #273 make this source-provider-agnostic
-  GitHubAccessTokenSecretId:
-    Description: The secretId of the GitHub Personal Access token stored in the Secrets Manager
+  ProviderType:
+    Description: For now, either GitHub or CodeCommit
     Type: String
-    Default: {{$.Source.GitHubPersonalAccessTokenSecretID}}
+    Default: {{.Source.ProviderName}}
+Conditions:
+  IsGitHub:
```

Comment:
For future concern what if we support more source? How do we make it extensible?

Reply:
Good question!
Refactored to use go templates instead of CFN conditionals (see comment below).